gradio/utils.py

```diff
@@ -352,3 +354,49 @@ def format_ner_list(input_string: str, ner_groups: Dict[str : str | int]):
 
     output.append((input_string[end:], None))
     return output
+
+
+class Http:
+    """
+    A class that shares an AsyncClient which sends and handles network requests.
+    Further different functionalities can be gathered within this class.
+    - Client configurations
+    - Response validation
+    - Support error types and error handling
+    - Client stop/restart
+    """
+
+    client = AsyncClient()
+
+    @classmethod
+    async def request(
```

Comment:
I think Http naming is so generic to do just making requests. You plan to do some validations (validation model/function). So there will be request related functions such as validate and parse. Creating a class for requests may be helpful for future improvements. 

This can be psoudo-implementated like the belowing.
```python
class Request:
    def __await__():
        return self.__run().__await__()
    def __init__(self,
                method: hdrs.METH_ALL,
                url: str,
                *,
                validation_model: Union[BaseModel, Type] = None,
                error_class: Type[Error] = Error,
                **kwargs):
        # Inıt request params.
        self.request = _create_request(...)
        # TODO: Init request related stuff...

    async def __run() -> Request:
        """
        A function to make request, validate and serialize
        """
        self._response = None
        self._validated_data = None
        self._exception = None

        try:
            # make client call and get response
            # Parse response
            # handle http errors
            # validate response ( with model and/or function)
            ...
        except Exception as e:
            self._exception = ...
        return self

# Making a request
response = await Request(hdrs.METH_PUT,
                         url=url,
                         json=payload,
                         timeout=timeout)
```

WDYT?

Reply:
@hibestil I think it looks very beautiful, would you like to implement this suggestion?

Reply:
Note that this branch has diverged from the main a lot, you might want to merge main or create another branch and move the work in this branch to there.